### PR TITLE
Moonbase setting improvements

### PR DESCRIPTION
**Pull request by xyzeva**
- Allow filter saving
- Add title into Config tab
- Don't allow setting changes on non-installed plugins

### CODE REVIEW

**Comment by redstonekasi** [comment]
This setting should not be in the config itself, it should be part of Moonbase's settings, maybe it'd be worth looking into giving plugins some kind of unexposed, unstructured settings storage? I can see several uses for that.

File: packages/core-extensions/src/moonbase/ui/config/index.tsx [hunk 1]
```diff
@@ -108,15 +109,39 @@ export default (require: WebpackRequireType) => {
...
+          </Text>
+        </div>
+
         <FormItem title="Repositories">
           <FormText className={Margins.marginBottom4}>
             A list of remote repositories to display extensions from
           </FormText>
           <ArrayFormItem config="repositories" />
         </FormItem>
-        <FormDivider className={FormClasses.dividerDefault} />
```
Comment (redstonekasi) [inline packages/core-extensions/src/moonbase/ui/config/index.tsx:119]: You killed the divider here, was that intended?

File: packages/core-extensions/src/moonbase/ui/config/index.tsx [hunk 2]
```diff
@@ -108,15 +109,39 @@ export default (require: WebpackRequireType) => {
...
+          className={`${TitleBarClasses.children} ${Margins.marginBottom20}`}
+        >
+          <Text
+            className={TitleBarClasses.titleWrapper}
+            variant="heading-lg/semibold"
+            tag="h2"
+          >
+            Config
+          </Text>
+        </div>
```
Comment (redstonekasi) [inline packages/core-extensions/src/moonbase/ui/config/index.tsx:126]: Also applies to the other page, these should not have the title bar classes, that was only for when the title bar also had the tabs (which I am very sad to see gone, but oh well), you should revert them to how the Moonbase header originally worked.

**Review by redstonekasi — CHANGES_REQUESTED** [review]
Didn't mention it in the earlier comment, but doing it this way will also pop up the settings changed notice when changing filters which is very bad.

File: packages/core-extensions/src/moonbase/ui/config/index.tsx (hunk 1 above)
Comment (xyzeva) [reply packages/core-extensions/src/moonbase/ui/config/index.tsx:119]: Yup, FormSwitch gives a divider

Comment (redstonekasi) [reply packages/core-extensions/src/moonbase/ui/config/index.tsx:119]: Yeah, but that's on the bottom.

Comment (xyzeva) [reply packages/core-extensions/src/moonbase/ui/config/index.tsx:119]: > Yeah, but that's on the bottom.

It showed up with a double divider for me when i tried, so /shrug

File: packages/types/src/config.ts [hunk 3]
```diff
@@ -4,6 +4,16 @@ export type Config = {
   devSearchPaths?: string[];
   loggerLevel?: string;
   patchAll?: boolean;
+  saveExtensionFilter: boolean;
```
Comment (NotNite) [inline packages/types/src/config.ts:7]: Don't store this in the global config; store it in the Moonbase config.

**Review by NotNite — CHANGES_REQUESTED** [review]

**Pull request merged**